planetiler-core/src/main/java/com/onthegomap/planetiler/util/FileUtils.java

```diff
@@ -314,4 +322,29 @@ public static void unzip(InputStream input, Path destDir) {
       throw new UncheckedIOException(e);
     }
   }
+
+  /**
+   * Unzips files from {@param zipPath} matching the glob {@param pattern} to {@param tmpLocation}. Extracted files will
+   * be deleted when the JVM exits.
+   *
+   * @return list of paths to extracted files
+   */
+  public static List<Path> unzipMatchingFiles(Path zipPath, String pattern, Path tmpLocation) throws IOException {
+    PathMatcher matcher = createGlobMatcher(zipPath.getFileSystem(), pattern);
+
+    try (var zipFs = FileSystems.newFileSystem(zipPath)) {
+      return walkFileSystemWithMatcher(zipFs, matcher)
+        .map(path -> {
+          try {
+            Path unzippedPath = tmpLocation.resolve(path.getFileName().toString());
+            Files.copy(path, unzippedPath, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Can we use the `unzip` utility here that has some safety checks built in?

Reply:
Would you want to extend `unzip` to only extract certain files, or unzip every file in the archive? The latter is probably fine by itself, but might conflict with the idea of pushing the extraction into the individual GeoPackageReader instead as you suggested here https://github.com/onthegomap/planetiler/pull/430#discussion_r1060510769

Reply:
Since we only wanted to extract a single path, most of the safety checks in `unzip` weren't relevant. I added a new helper which preserves the max size check though